src/weapons.cpp

```diff
@@ -132,6 +135,19 @@ bool Weapons::registerEvent(Event_ptr event, const pugi::xml_node&)
 	return result.second;
 }
 
+bool Weapons::registerLuaEvent(Weapon* event)
+{
+	Event_ptr w{ event };
+	Weapon* weapon = static_cast<Weapon*>(w.release()); //event is guaranteed to be a Weapon
```

Comment:
Why are you creating an `Event_ptr` from a `Weapon` pointer then immediately converting that `Event_ptr` back into a `Weapon` pointer?

Reply:
yea forgot to sort this out, thanks for pointing it out.
I'll just create a `Weapon_ptr` and make use of it directly.

Reply:
fixed in [9148138](https://github.com/otland/forgottenserver/pull/2571/commits/91481384539c767c1ab0570a52f8c011a022bb1b)